Xamarin.Essentials/WebAuthenticator/WebAuthenticator.android.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Threading.Tasks;
+using Android.Content;
+using Android.Support.CustomTabs;
+
+namespace Xamarin.Essentials
+{
+    public partial class WebAuthenticator
+    {
+        static TaskCompletionSource<AuthResult> tcsResponse = null;
+
+        static Uri uri = null;
+
+        static CustomTabsActivityManager CustomTabsActivityManager { get; set; }
+
+        static Uri RedirectUri { get; set; }
+
+        internal static Task<AuthResult> ResponseTask
+            => tcsResponse?.Task;
+
+        internal static bool OnResume(Intent intent)
+        {
+            // If we aren't waiting on a task, don't handle the url
+            if (tcsResponse?.Task?.IsCompleted ?? true)
+                return false;
+
+            if (intent == null)
+            {
+                tcsResponse.TrySetCanceled();
```

Comment:
Just so I understand, if i resume from anything but the auth activity, it is a cancel?

Reply:
So the idea is that if the user cancel's out of the auth flow, we have no way to know other than the previous activity being resumed before we ever get a callback (which will never happen).  When the previous activity resumes it calls this OnResume method just like the callback activity would, but in this case it passes a null value, and so we know that it was canceled.